pySDC/core/ConvergenceController.py

```diff
@@ -169,29 +168,12 @@ def setup_status_variables(self, controller, **kwargs):
 
         Args:
             controller (pySDC.Controller): The controller
-            reset (bool): Whether the function is called for the first time or to reset
 
         Returns:
             None
         """
         return None
 
-    def reset_buffers(self, controller, **kwargs):
-        """
-        Buffers refer to variables used across multiple steps that are stored in the convergence controller classes to
-        do communication. These can be reset in order to make sure the value was freshly communicated rather than
-        reused.
-
-        This function is called both at the end of instantiating the controller, as well as after each iteration.
-
-        Args:
-            controller (pySDC.Controller): The controller
-
-        Returns:
-            None
-        """
-        pass
-
     def reset_buffers_nonMPI(self, controller, **kwargs):
         """
```

Comment:
Do you need this function?

Reply:
The non MPI one I would like to keep. It doesn't really add much functionality, but it's meant to prevent bugs. By reinitialising the buffers for simulating sending between each step you make sure you don't accidentally use data sent in the last step. Also, this is called during initialisation to setup these buffers before the run.

The MPI one, however is not needed because there is only real communication.